### DISCUSSION TITLE
.TXT file attachment to inbound email is ignored in django-helpdesk 1.0

### DISCUSSION THREAD

**Discussion by trainor1** [django-helpdesk/django-helpdesk#1126] [Q&A]
Hi All,

While testing django-helpdesk 1.0 as the solution to my email ingest problems, I discovered that emails with .TXT attachments were being processed into tickets without the .TXT attachment having been added to the ticket.  This is happening despite .txt being listed as the first entry in the VALID_EXTENSIONS list in settings.

I also tested .PDF and .ZIP.  These are my other common use cases.  These worked OK.  I did not test other extensions in my VALID_EXTENSIONS list.

Best,

Kevin

**Comment by uhurusurfa** [comment]
I have run a trest with a simple .txt extension and it worked for me.
Can you provide the raw email that is failing to be processed for you to see if there is an unaccounted for MIME structure?

**Reply by trainor1** [reply]
I ran the test again here and got the same results.  Here are some files for you to look at:

[41966748.mime.txt](https://github.com/django-helpdesk/django-helpdesk/files/13199477/41966748.mime.txt)
[wet_birds.txt](https://github.com/django-helpdesk/django-helpdesk/files/13199481/wet_birds.txt)
<img width="1420" alt="ticket_without_attachement" src="https://github.com/django-helpdesk/django-helpdesk/assets/5059944/489adab9-7a78-4c94-a27c-efb85c3b6a3f">

**Reply by uhurusurfa** [reply]
Can you confirm that the email you attached to this ticket was actually in the INBOX that is being processed by Helpdesk when you saved the raw text version?

I have fed the exact file you sent into the email parser and it correctly extracted the atachment using a pure vanilla deployment of the latest Helpdesk 1.0.0 code base and given that your tests show that .pdf and .
The only mechanism to exclude specific attachments is when the attachment record is saved to the database. The email parser does not care about extensions, it simply extracts every part it finds into a list of files to be attached and then the save event calls  the validator when the record is saved to check if that attachment is allowed to be saved. That functionality has not changed in quite a long time now.

If it were the VALID_EXTENSI"ONS causing the file to be ignored there would be an error message starting with "**Unsupported file extension:**" in the logs.

At this point we need to be 100% certain that the email you attached this ticket is the exact content that would be processed by the email parser before trying to figure out next steps to debug this.

**Reply by trainor1** [reply]
The email that I attached to this discussion was indeed in the inbox of the email account.  I retrieved the text that I sent to you from my Sent Mail folder.  Django Helpdesk read the email and created a ticket.  That ticket did not include the .TXT file attachment.

**Reply by uhurusurfa** [reply]
So whilst I understand that you feel that because the MIME text you obtained from your sent folder should be exactly what is received by the email server that Helpdesk is getting emails from via the INBOX, it is not necessarily going to be in the same format because email servers add headers and thereby change what is actually received in the INBOX.

In order to completely rule out some structural change in the MIME that is somehow the cause of the problem, can you either:

1.   Disable the invoking of the **get_email** task and then extract the raw message from the INBOX
2. Add a print statement to helpdesk/email.py after line 951:
      print(f"MIME TEXT\n###############\n{message}\n###############\n")
      
      Then run the **get_email** task from the command line:
      /path/to/helpdesksite/manage.py get_email
      ... and upload the MIME text that is printed to the console.

**Reply by trainor1** [reply]
This is a new test.  I saved the email file while it was in the inbox on Microsoft Office 365.  Here is the email:

[Ticket to Test .TXT Attachment.eml.txt](https://github.com/django-helpdesk/django-helpdesk/files/13221173/Ticket.to.Test.TXT.Attachment.eml.txt)

Here is a link to the ticket that was created:

If you wish to view this ticket online to provide further information, attach files or view recent updates, you can visit https://trainorhelpdesk.ligent.net/view/?ticket=test-course-176&email=ktrainor@ligent.net&key=6dc287c1-2324-4174-ba62-2afcc1aa77f2.

I hope this helps.

**Reply by uhurusurfa** [reply]
Unfortunately the parser still correctly extracts the file as an attachment in my tests.
I have attached an enhanced version of the email.py file that will create a log file in the working directory when you startup up Django. (it is renamed with a .txt extension so I could attach it here so please rename back to email.py)

Please replace the file in your installed app with this file. You should find the file in: <helpdesk app install dir>/helpdesk/email.py

Once you have sent another test email to the server (ensure the scheduled calling of the get_email task is disabled) , run the get_email task from the command line and there should be a file created in the directory where you ran the get_email task from named "get_email_debug.log"
Please upload that file.
[email.txt](https://github.com/django-helpdesk/django-helpdesk/files/13249074/email.txt)

**Reply by trainor1** [reply]
I appreciate your work on this.  I have been distracted by my day job and unable to follow up on my end.  But, I hope to get back to it soon.  It occurs to me that I should make sure that I have the most up-to-date copy of the software installed and that I should try to reproduce the symptoms using that software.  Here are a few questions related to making sure that I have the up-to-date software:

1. I have been getting my django-helpdesk from PyPI.  Is that sufficient?
2. I have not been doing anything special to make sure that I am running a current version of the get_mail command.  Is that updated when I update the django-helpdesk package that is imported into my project?  Or, do I need to do something further to apply updates to that function?

Thanks again for your help.

Best,

Kevin

**Reply by uhurusurfa** [reply]
The email.txt file I attached above is a special version that adds a whole lot of logging to try to get to the bottom of this.
Functionally it is no diffferent to the email.py.

Using PyPi is the correct route to get the latest version but after updating you should replace email.py with the debug version to debug this issue.

**Comment by uhurusurfa** [comment]
No fuirther feedback so closing
